routers/api/v1/repo/project.go

```diff
@@ -0,0 +1,240 @@
+package repo
+
+import (
+	"fmt"
+	"net/http"
+
+	project_model "code.gitea.io/gitea/models/project"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/convert"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/web"
+)
+
+func GetProject(ctx *context.APIContext) {
+	// swagger:operation GET /projects/{id} project projectGetProject
+	// ---
+	// summary: Get project
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the project
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Project"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	project, err := project_model.GetProjectByID(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		if project_model.IsErrProjectNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetProjectByID", err)
+		}
+		return
+	}
+	ctx.JSON(http.StatusOK, convert.ToAPIProject(project))
+}
+
+func UpdateProject(ctx *context.APIContext) {
+	// swagger:operation PATCH /projects/{id} project projectUpdateProject
+	// ---
+	// summary: Update project
+	// produces:
+	// - application/json
+	// consumes:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the project
+	//   type: string
+	//   required: true
+	// - name: project
+	//   in: body
+	//   required: true
+	//   schema: { "$ref": "#/definitions/UpdateProjectPayload" }
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Project"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	form := web.GetForm(ctx).(*api.UpdateProjectPayload)
+	project, err := project_model.GetProjectByID(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		if project_model.IsErrProjectNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetProjectByID", err)
+		}
+		return
+	}
+	if form.Title != project.Title {
+		project.Title = form.Title
+	}
+	if form.Description != project.Description {
+		project.Description = form.Description
+	}
+	err = project_model.UpdateProject(ctx, project)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "UpdateProject", err)
+		return
+	}
+
+	ctx.JSON(http.StatusOK, project)
+}
+
+func DeleteProject(ctx *context.APIContext) {
+	// swagger:operation DELETE /projects/{id} project projectDeleteProject
+	// ---
+	// summary: Delete project
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the project
+	//   type: string
+	//   required: true
+	// responses:
+	//   "204":
+	//     "description": "Deleted the project"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	err := project_model.DeleteProjectByID(ctx, ctx.ParamsInt64(":id"))
```

Comment:
As above.

Reply:
what error should we return in this case, Unauthorized, BadRequest or NotFound ?